src/libs/ReportUtils.ts

```diff
@@ -1415,12 +1415,28 @@ function isJoinRequestInAdminRoom(report: OnyxEntry<Report>): boolean {
     return ReportActionsUtils.isActionableJoinRequestPending(report.reportID);
 }
 
+/**
+ * Checks if the user has auditor permission in the provided report
+ */
+function isAuditor(report: OnyxEntry<Report>): boolean {
```

Comment:
Okay so I think this is slightly wrong - I think we should check for `isPolicyAuditor` first and THEN check report permissions afterwards. Otherwise the following test steps seem to occur:

1. Create a workspace
2. Add two members, then make one an admin and one an auditor
3. Assign a task to each of them in the `#admins` chat
4. Log in as the auditor
5. The auditor will be able to edit both tasks.

I looked at Onyx and the auditor's report permissions for the task assigned to the admin is `read, write, share`. I'm honestly not sure why that is, but it seems like if we checked `isPolicyAuditor` first then we wouldn't get to that point.

What do you think? Would this cause any issues I'm not aware of?

Reply:
Follow up - let me know how you were testing it before if it was seeming to work then? Maybe I did something wrong - or maybe there's a flow (maybe the change role isn't working or something?) that we need to adjust

Reply:
Thanks let me take a look

Reply:
Yes, that's pretty weird that on the admin assigned task, when I viewed it as an auditor the permissions are read, write, share which makes `isAuditor` return false. Meanwhile it should be empty (as I checked/ tested before 😂).

Switch the `PolicyUtils.isPolicyAuditor(policy);` to the top will make it works because we will use the role to check it. Nevermind, let me update the code.

Thanks for testing it! 💯

Reply:
Updated!